docs/database/database-spans.md

```diff
@@ -85,6 +87,56 @@ and SHOULD adhere to one of the following values, provided they are accessible:
 If a corresponding `{target}` value is not available for a specific operation, the instrumentation SHOULD omit the `{target}`.
 For example, for an operation describing SQL query on an anonymous table like `SELECT * FROM (SELECT * FROM table) t`, span name should be `SELECT`.
 
+## Status
+
+[Span Status Code][SpanStatus] MUST be left unset if the operation has ended without any errors.
+
+Instrumentation SHOULD consider the operation as failed if any of the following is true:
+
+- the `db.response.status_code` value indicates an error
+
+  > [!NOTE]
+  > The classification of status code as an error depends on the context.
+  > For example, a SQL STATE `02000` (`no_data`) indicates an error when the application
+  > expected the data to be available. However, it is not an error when the
+  > application is simply checking whether the data exists.
+  >
+  > Instrumentations that have additional context about a specific operation MAY use
+  > this context to set the span status more precisely.
+  > Instrumentations that don't have any additional context MUST follow the
+  > guidelines in this section.
+
+- an exception is thrown by the instrumented method call
+- the instrumented method returns an error in another way
+
+When the operation ends with an error, instrumentation:
+
+- SHOULD set the span status code to `Error`
+- SHOULD set the `error.type` attribute
+- SHOULD set the span status description when it has additional information
+  about the error that aligns with [Span Status Description][SpanStatus].
+
+  It's NOT RECOMMENDED to duplicate `db.response.status_code` or `error.type`
+  in span status description.
+
+  When the operation fails with an exception, the span status description SHOULD be set to
+  the exception message.
+
+### Recording exception events
+
+**Status**: [Experimental][DocumentStatus]
+
+When the operation fails with an exception, instrumentation SHOULD record
+an [exception event](../exceptions/exceptions-spans.md) by default if and only if the
+span being recorded is a local root span (does not have a local parent).
+
+> [!NOTE]
+> Exception stack traces could be very long and are expensive to capture and store.
+> Exceptions which are not handled by instrumented libraries are likely to be handled
+> and logged by the caller.
+> Exceptions that are not handled will be recorded by the outermost (local root)
+> instrumentation such as HTTP or gRPC server.
```

Comment:
let's ask for feedback on this point from the Specification SIG before we merge

Reply:
I think this is a directional shift in logging exceptions.

I see two things here:

1. We still can't determine "layers" of instrumentation, i.e. we *can't* know if someone else will log the exception or not.
2. This violates the previously recorded wisdom of how to handle exceptions and stack traces.


My opinion:

This is the right direction to go.  We *should* assume that there is "outer" instrumentation that logs/catches exceptions, particularly for database "client" spans.  I.e. we may be able to blanketly state that "client" spans should never log unhandled exceptions and that we encourage "outer" instrumentation that will (e.g. server spans).

Reply:
I added to next week's spec SIG agenda

Reply:
Feedback from the spec SIG was that this sounds like a good default behavior, but we should make it configurable.

Some options for making it configurable:

(A) handle this all in the SDK:
* new configuration added to [TracerConfig](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#tracerconfig)
* [RecordException](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#record-exception) behavior could depend on this new configuration in TracerConfig

but I don't think we can change the default behavior of the SDK.

(B) Another option could be to introduce "trace advice" (similar to metric advice) that would allow a given instrumentation the ability to opt-in to this new behavior.

(C) Another option could be:
* new configuration added to [TracerConfig](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#tracerconfig)
* instead of changing the (existing) behavior of RecordException, ask instrumentations to check this configuration value before calling RecordException

(D) Another option could be:
* Introduce a standard "database" instrumentation configuration property (similar to the standard "http" configuration properties that we already have: https://github.com/open-telemetry/opentelemetry-configuration/blob/e1f89ef021f3c8e000d05e6b6e53bc4a2372fe52/examples/kitchen-sink.yaml#L396-L400)
* ask instrumentations to check this configuration value before calling RecordException
* we may also want to be able to override this at individual instrumentation level somehow
 
(E) Or even:
* Introduce a global instrumentation configuration property that covers all instrumentations
* ask instrumentations to check this configuration value before calling RecordException
* we may also want to be able to override this at individual instrumentation level somehow

Probably we should spin this out into a separate issue.

cc @jack-berg

Reply:
Thanks for the notes!

Since we don't usually specify how instrumentations should configure things, I'd prefer to tackle it separately. 
It's definitely a problem that we should fix in the https://github.com/open-telemetry/semantic-conventions/issues/1536 and https://github.com/open-telemetry/semantic-conventions/issues/1450